src/internal/connector/graph/service.go

```diff
@@ -160,6 +172,40 @@ func HTTPClient(opts ...option) *http.Client {
 	return httpClient
 }
 
+// GetDefaultMiddlewares creates a new default set of middlewares for the Kiota request adapter
+func GetMiddlewares() []khttp.Middleware {
+	return []khttp.Middleware{
+		// khttp.NewRetryHandler(),
```

Comment:
We should probably keep the khttp retry handler in place.  Continue letting it handle retries in the way that it prefers (ex, handling throttling, etc), and only concern ourselves with the handling that's left over.

Reply:
We can do that but the issue is - we won't be able to control the `no. of retries` or `delays` in that case. And have to go with default khttp values.